plugins/thread.py

```diff
@@ -117,12 +137,43 @@ def run(self):
         # copy to & open output dir
         try:
             shutil.copytree(temp_dir, output_path, dirs_exist_ok=True)
-            webbrowser.open("file://%s" % (output_path))
+            if self.openBrowser:
+                webbrowser.open("file://%s" % (output_path))
             shutil.rmtree(temp_dir)
-        except Exception as e: 
-            webbrowser.open("file://%s" % (temp_dir))
+        except Exception as e:
+            if self.openBrowser:
+                webbrowser.open("file://%s" % (temp_dir))
 
-        self.progress(-1)
+        if self.wx is None: 
+            self.progress(100)
+        else:
+            self.progress(-1)
 
     def progress(self, percent):
-        wx.PostEvent(self.wx, StatusEvent(percent))
+        def cliBar(percent, prefix = '', suffix = '', decimals = 1, length = 100, fill = '█', printEnd = "\r"):
```

Comment:
What do you think about moving this function to the utils module and make it generate the progress bar string instead of printing it. than it can be used in the progress function instead of nesting inside it.